pkg/traceql/engine_metrics_compare.go

```diff
@@ -167,6 +167,7 @@ func (m *MetricsCompare) observe(span Span) {
 		}
 		totals[i]++
 	})
+	return len(m.seriesAgg.Results())
```

Comment:
can we get tests on these all these series aggregators observe's returning total series?